### PR TITLE
Agent kube projected token

### PR DESCRIPTION
**Pull request by catsby**
This PR adds support for using non-default ServiceAccounts when using Vault Agent and auto-auth on Kubernetes by adding an additional configuration option "token_path".

### CODE REVIEW

File: command/agent/auth/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -54,14 +74,15 @@ func NewKubernetesAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 
 func (k *kubernetesMethod) Authenticate(ctx context.Context, client *api.Client) (string, map[string]interface{}, error) {
 	k.logger.Trace("beginning authentication")
-	content, err := ioutil.ReadFile(serviceAccountFile)
+
+	jwtString, err := k.readJWT()
 	if err != nil {
-		log.Fatal(err)
+		return "", nil, errwrap.Wrapf("error reading JWT with Kubernetes Auth: {{err}}", err)
```
Comment (catsby) [inline command/agent/auth/kubernetes/kubernetes.go]: This needs good review - the previous behavior was to log the error and exit completely, which seemed to be excessive.

File: command/agent/auth/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -22,8 +24,17 @@ type kubernetesMethod struct {
 	mountPath string
 
 	role string
+
+	// tokenPath is an optional path to a projected service account token inside
+	// the pod, for use instead of the default service account token.
+	tokenPath string
+
+	// jwtData is a ReaderCloser used to inject a ReadCloser for mocking tests.
```
Comment (jefferai) [inline command/agent/auth/kubernetes/kubernetes.go]: s/ReaderCloser/ReadCloser?

File: command/agent/auth/kubernetes/kubernetes.go (hunk 1 above)
Comment (jefferai) [reply command/agent/auth/kubernetes/kubernetes.go]: That was almost certainly something accidentally left in from debug sessions when it was initially being written, so this looks good.

File: command/agent/auth/kubernetes/kubernetes.go (hunk 2 above)
Comment (catsby) [reply command/agent/auth/kubernetes/kubernetes.go]: Fixed in b24487aee

**Pull request merged**